install/helm/gloo/generate/values.go

```diff
@@ -224,6 +224,7 @@ type GlooDeployment struct {
 type Discovery struct {
 	Deployment     *DiscoveryDeployment `json:"deployment,omitempty"`
 	FdsMode        *string              `json:"fdsMode,omitempty" desc:"mode for function discovery (blacklist or whitelist). See more info in the settings docs"`
+	UdsEnabled     *bool                `json:"udsEnabled,omitempty" desc:"enable upstream discovery service (default true)."`
```

Comment:
Per https://github.com/solo-io/gloo/pull/5546#discussion_r745082615 docs would like this in `sentence-case`

```suggestion
	UdsEnabled     *bool                `json:"udsEnabled,omitempty" desc:"Enable upstream discovery service (default true)."`
```

Reply:
Also question do we foresee wanting to turn this into something like we have for fds? Namely for future proofing do we think there may be a next request to set this to something like allowlist or blocklist?

Reply:
I like your thought process. It's certainly possible we may want more than just an on/off switch for UDS in the future.  Which sounds more reasonable to you?

1. Replace `UdsEnabled` here with a reference to a `type UdsOptions struct` which in turn holds an `Enabled` field. This struct can be added to in the future if needed - more or less just for organization.
2. Replace `UdsEnabled` here with a `UdsMode` enum (analogous to `FdsMode`).  The issue I see here is that we need to have a default for the enum. The sane approach would be to have `UdsMode` default to `ENABLED` (maintaining the current behavior). But then that means we need a `UdsMode.DISABLED` enum value as well to support this feature.  Now if we want to elaborate this in the future, we would have `UdsMode = { ENABLED, DISABLED, WHITELIST, BLACKLIST }`, which is pretty ugly.

I'm in favor of option 1. If we have a primary on/off switch for UDS, we can still introduce new UDS behavior by adding fields to the `UdsOptions` struct. If `UdsOptions.Enabled = false`, all other fields in that struct would be ignored.

Reply:
I really like your point about approach two its nice to call out that the default behavior would have to be different from Fds for legacy default enablements.

Given this I think you are right that approach 1 is the better approach for future proofing so I think that as long as we forsee it being likely that we add power to options then we take this approach and if we are very confident that we wont add anything in the future then we go with the bool as is.